src/lib/render.ts

```diff
@@ -36,11 +37,16 @@ export const parts = new WeakMap<Node, NodePart>();
 export function render(
     result: TemplateResult,
     container: Element|DocumentFragment,
-    templateFactory: TemplateFactory = defaultTemplateFactory) {
+    options?: Partial<RenderOptions>) {
   let part = parts.get(container);
   if (part === undefined) {
     removeNodes(container, container.firstChild);
-    parts.set(container, part = new NodePart(templateFactory));
+    parts.set(
+        container, part = new NodePart({
+                     eventContext: options && options.eventContext,
+                     templateFactory: options && options.templateFactory ||
+                         defaultTemplateFactory,
```

Comment:
What was your rationale building up the default options here, vs. having their usage points fall back to defaults if not provided in `options`? Avoiding the need to  undefined-test the usage points? Seems like a good tradeoff for perf, just wanted to confirm since it means this spot needs to know about every option, which is kind of a bummer.

Reply:
Yeah, it was to avoid the checks and be able to set the default template factory here rather than that being determined at the use site. I think I can make this better though with spread so other options pass through.

It occurs to me though that we really never want these options to change, because they're saved into Parts when parts are created. If options changed in-between renders, Parts may have different options. One way around that is to remove options altogether from `render()`, and instead vend a render factory that takes options. Then we don't create the temporary options object either. WDYT?